code_curation2_submissions2/Julian_clustering_pipeline.R

```diff
@@ -0,0 +1,246 @@
+########################
+## CV new pipeline
+########################
+# (c) JU_14062019
+
+# This script runs literature informed optimisation pipeline for clustering. 
+# Conceptual and methodological aspects can be found under:
+# '/Users/Julian/Desktop/projects/COG_SUB_01062019/manuscript/'
+
+
+# empty workspace, define general project path, load project functions
+rm(list=ls())
+project_path <- '/Users/Julian/Desktop/projects/COG_SUB_01062019'
+setwd(paste0(project_path, '/project_functions/'))
+sapply(list.files(), source)
+
+# define project dirs and load/install packages needed
+activate_pkg(list_of_packages = c('e1071', 'tidyr', 'fpc', 'rlist', 'caret', 'UBL', 'devtools', 'ggplot2', 'VIM', 'doBy')) # installs all relevant packages 
+COG_SUB_paths(project_path, raw = 0, prepro = 1, clust = 1, MRI = 1)
+
+
+####################################################################################################################
+
+###############
+## data entry
+##############
+
+# load in (uncorrected) patient data
+setwd(prepro_results_dir)
+data <- read.csv('data_no_imp.csv', header = TRUE, sep = ',', dec = '.')
+
+# add edu variable to patient data
+setwd(prepro_data_dir1)
+data_edu_ROP <- read.csv('DataQuery_LKI_GA_04_2018_JU_Data_all_07-Nov-2018.csv', header = TRUE, sep = ',', dec = '.', na.strings='NaN')[, c('PSN', 'DEMOG_T0T1T2_31AA_EducationYears_T0')]
+setwd(prepro_data_dir2)
+data_edu_HC <- read.csv('DataQuery_LKI_JU_01_2019_Data_all_01-Feb-2019.csv', header = TRUE, sep = ',', dec = '.', na.strings='NaN')[, c('PSN', 'DEMOG_T0T1T2_31AA_EducationYears_T0')]
+data <- merge(data, data_edu_ROP, by='PSN', all.x=T)
+data_HC <- merge(data_HC, data_edu_HC, by='PSN', all.x=T)
+names(data)[names(data)=='DEMOG_T0T1T2_31AA_EducationYears_T0'] <- 'Edu_years'
+names(data_HC)[names(data_HC)=='DEMOG_T0T1T2_31AA_EducationYears_T0'] <- 'Edu_years'
+data[is.na(data$Edu_years),'Edu_years'] <- median(data$Edu_years, na.rm=T)
+data_HC[is.na(data_HC$Edu_years),'Edu_years'] <- median(data_HC$Edu_years, na.rm=T)
+
+# variables indices for patient data
+neuropsy_ind <- c(49:53, 58:131, 136:139) # indices of neuropsy vars
+clinical_ind <- c(5:10, 132:134 ,41:48)   # indices of clinical vars 
+
+# load in imputed healthy control data
+data_HC <- read.csv('data_preprocessed_HC_2019-04-23.csv', header = TRUE, sep = ',', dec = '.')
+colnames(data_HC)[colnames(data_HC)=='SEX_GAF_T0_Screening'] <- 'SEX_T0' 
+
+# variable indices for healthy control data
+np_ind_HC  <- c(39:43, 48:121, 126:129) # indices of NP vars in healthy controls
+
+###############################
+## specifications of pipeline
+###############################
+
+num_perm <- 5 # number of subsettings of the data set 
+perm <- 1:num_perm # permutation range
+perc <- 0.75  # amount of cases by which data is subset in percentage
+num_cluster <- 2:5 # numbers of cluster to be investigated
+clust_alg <- c('kmeans', 'hierach')  # clustering algorithms: 'kmeans', 'hierach' are allowed

```

Comment:
Are these User defined variables or constants?
I either case, they should be placed at the beginning of the script.